web3/types.py

```diff
@@ -112,7 +112,20 @@ class ABIFunction(TypedDict, total=False):
     type: Literal["function", "constructor", "fallback", "receive"]
 
 
-ABIElement = Union[ABIFunction, ABIEvent]
+class ABIError(TypedDict):
+    """
+    TypedDict representing the `ABI` for an error.
+    """
+
+    type: Literal["error"]
+    """Type of the error."""
+    name: str
+    """Name of the error."""
+    inputs: NotRequired[Sequence["ABIFunctionParams"]]
```

Comment:
It doesn't feel fantastic to use `ABIFunctionParams` here. I could be swayed to let it slide but it feels pretty minimal to also introduce the new `ABIComponent` to v6 and use it here instead? Thoughts on that?

Reply:
No complaints from me, it is better to use the correct types from the start. I considered using it but left it out to do the bare minimum change.